async functions return bool so in C returning null (a pointer) counts as a
int-conversion error in new clang/gcc
--
--- a/src/libvirt-broker.vala
+++ b/src/libvirt-broker.vala
@@ -26,7 +26,7 @@
 
     public async LibvirtMachine add_domain (CollectionSource source, GVir.Connection connection, GVir.Domain domain)
                                             throws GLib.Error {
-        return_val_if_fail (broker != null, null);
+        return_val_if_fail (broker != null, 0);
 
         if (pending_domains.find (domain) != null) {
             // Already being added asychronously
